lstchain/calib/camera/time_sampling_correction.py

```diff
@@ -0,0 +1,118 @@
+import h5py
+import numpy as np
+
+
+from ctapipe.core import Component
+from ctapipe.core.traits import Unicode
+
+__all__ = [
+    'TimeSamplingCorrection',
+    ]
+
+
+
+class TimeSamplingCorrection(Component):
+    """
+        The PulseTimeCorrection class to correct time pulse
+        using Fourier series expansion.
+    """
+
+    time_sampling_correction_path = Unicode(
+        '',
+        help='Path to the waveform sampling correction file',
+        allow_none = True,
+    ).tag(config=True)
+
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+
+
+        self.time_sampling_coefficients = None
+
+        self.load_sampling_coefficient_file()
+
+    def load_sampling_coefficient_file(self):
+        """
+            Function to load sampling coefficient file.
+        """
+
+        try:
+            with h5py.File(self.time_sampling_correction_path, 'r') as hf:
+                self.time_sampling_coefficients = np.array(hf['sampling_interval_coefficient'])
+        except:
+            self.log.error(f"Problem in reading sampling coefficient file {self.time_sampling_correction_path}")
+
+
+    def get_corrections(self, event, telid):
+        """
+        Get the time/charge sampling corrections for one event and one telescope
+
+        Parameters
+        ----------
+        event: general event container
+        telid: id of the telescope
+
+        Return
+        ------
+        sampling_corrections: np.array (n_gains, n_pixels, n_samples) with the correction factors
+        """
+
+        n_gains = 2
+        n_modules = 265  # number of modules in LST's camera.
+        n_pixel_per_module = 7  # number of pixels per one module.
+        size_drs4 = 1024
+        n_pixels = 1855
+        roi = 36
+
+        # shift the first capacitor to the value used in the r1
+        r1_roi_start = 3
+
+        fc_all = event.lst.tel[telid].evt.first_capacitor_id
+
+        # get the first capacitor per pixel and gain
+        fc = np.zeros((n_gains, n_pixels), dtype=np.int16)
+
+        for k in range(n_modules):
+            for n in range(n_pixel_per_module):
+                fc[0][k * 7 + n] = fc_all[8 * k + n // 2]
```

Comment:
You define `n_pixel_per_module` above but then use hard coded magic numbers here (7 and 8). 

What is `8`? I assume the number of channels on a DRS4 chip?

Reply:
Changes